cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceContextProvider.java

```diff
@@ -90,6 +90,12 @@ public synchronized BasicMapReduceContext get() {
     return context;
   }
 
+  public synchronized void stop() {
```

Comment:
It sounds like a bad idea to have provider or builder to method like stop() or finish() or close(). Can you open a JIRA for refactoring that?